src/open-uri.c

```diff
@@ -171,18 +163,33 @@ get_latest_choice_info (const char *app_id,
   *latest_id = choice_id;
   *latest_count = choice_count;
   *latest_threshold = choice_threshold;
-  *always_ask = choice_always_ask;
 
   return (choice_id != NULL);
 }
 
 static gboolean
 is_sandboxed (GDesktopAppInfo *info)
 {
-  g_autofree char *exec;
+  g_autofree char *flatpak = NULL;
+
+  flatpak = g_desktop_app_info_get_string (G_DESKTOP_APP_INFO (info), "X-Flatpak");
+
+  return flatpak != NULL;
+}
+
+static char *
+get_actual_app_id (GAppInfo *info)
+{
+  g_autofree char *flatpak = NULL;
+  const char *desktop_id;
+
+  flatpak = g_desktop_app_info_get_string (G_DESKTOP_APP_INFO (info), "X-Flatpak");
+  desktop_id = g_app_info_get_id (info);
 
-  exec = g_desktop_app_info_get_string (info, G_KEY_FILE_DESKTOP_KEY_EXEC);
-  return strstr (exec, "flatpak run ") != NULL;
+  if (flatpak)
+    return g_steal_pointer (&flatpak);
+  else
+    return g_strndup (desktop_id, strlen (desktop_id) - strlen (".desktop"));
```

Comment:
This is not necessarily right. Apps are allowed to use "sub-ids" of the app id. And apps like libreoffice does this for its multiple apps.

Reply:
Oh, i see this is the fallback. Sorry.

Reply:
I wonder though if we should support such split desktop files to (optionally) specify their main id, so that the same permissions would be used for the flatpak and the host version of the app.

Reply:
It would have to be optional to support the 'private browser' use case that @TingPing mentioned above.

Reply:
hmm, so to support application sub-ids here, I think I need to change course, and not normalize things to the flatpak ID. There can be multiple sub-ids, and they can be launched individually, so we want to let the app chooser present them and pass a sub-id back to the frontend to launch. And I guess that also implies we need to allow sub-ids in the permission store, so we can count how often the user has chosen the same sub-id.